src/controllers/bmdashboard/bmInventoryTypeController.js

```diff
@@ -363,35 +362,110 @@ function bmInventoryTypeController(InvType, MatType, ConsType, ReusType, ToolTyp
+    const { unit, category = 'Material' } = req.body;
+    if (typeof unit !== 'string' || unit.length === 0) {
+      res.status(400).json('Invalid unit');
+      return;
+    }
+
     try {
-      const { invtypeId } = req.params;
-      const { name, unit } = req.body;
+      // read JSON file and parse it into an array
+      const unitsJSON = await fsPromises.readFile(filepath, { encoding: 'utf8' });
+      const unitsArray = JSON.parse(unitsJSON);
 
-      const updateData = {};
+      // append new unit into array
+      unitsArray.push({ unit, category });
 
-      if (name) {
-        updateData.name = name;
-      }
+      // save updated array into JSON file and rend it back
+      await fsPromises.writeFile(filepath, JSON.stringify(unitsArray, null, ' '));
+
+      res.status(201).send(unitsArray);
+    } catch (err) {
+      res.status(500).send(err);
+      console.error(err);
+    }
+  };
+
+  const deleteInvUnit = async (req, res) => {
+    const { unit } = req.body;
+    if (typeof unit !== 'string' || unit.length === 0) {
+      res.status(400).json('Invalid unit');
+      return;
+    }
 
-      if (unit) {
-        updateData.unit = unit;
+    try {
+      // read JSON file and parse it into an array
+      const unitsJSON = await fsPromises.readFile(filepath, { encoding: 'utf8' });
+      const unitsArray = JSON.parse(unitsJSON);
+
+      // if unit does not exist, send err response
+      const index = unitsArray.findIndex((unitObject) => unitObject.unit === unit);
+      if (index === -1) {
+        res.status(400).json('Unit does not exist');
+        return;
       }
 
-      const updatedInvType = await InvType.findByIdAndUpdate(invtypeId, updateData, {
-        new: true,
-        runValidators: true,
-      });
+      // otherwise, remove unit
+      const filteredUnits = unitsArray.filter((unitObject) => unitObject.unit !== unit);
 
+      // save updated array into JSON file and rend it back
+      await fsPromises.writeFile(filepath, JSON.stringify(filteredUnits, null, ' '));
+      res.status(200).send(filteredUnits);
+    } catch (err) {
+      res.status(500).send(err);
+      console.error(err);
+    }
+  };
+
+  const updateSingleInvType = async (req, res) => {
+    const { type, invtypeId } = req.params;
+    const { name, description } = req.body;
+
+    // send back errors if required fields are missing
+    if (name?.length === 0 || description?.length === 0) {
+      res.status(400).json({ error: 'Name and description are required.' });
+      return;
+    }
+
+    try {
+      // find invType by id, and update name, description
+      const updatedInvType = await InvType.findByIdAndUpdate(
+        invtypeId,
+        { name, description },
+        { new: true, runValidators: true },
+      );
       if (!updatedInvType) {
-        return res.status(404).json({ error: 'invType Material not found check Id' });
+        res.status(404).json({ error: 'invTypeId does not exist' });
```

Comment:
It would be better to use general error message instead of using variable name "invTypeId" in error message.